models/engine.go

```diff
@@ -16,13 +21,34 @@ type Engine interface {
 
 var _ Engine = &RestartableEngine{}
 
+func NewRestartableEngine(engine Engine, retries int, logger zerolog.Logger) *RestartableEngine {
```

Comment:
can you add a comment noting that if retires is less than 3, what the behavior will be

Reply:
lol